be/src/olap/column_vector.h

```diff
@@ -51,6 +51,37 @@ class DataBuffer {
     void resize(size_t _size);
 };
 
+template <class T>
+DataBuffer<T>::DataBuffer(size_t new_size) : buf(nullptr), current_size(0), current_capacity(0) {
+    resize(new_size);
+}
+
+template <class T>
+DataBuffer<T>::~DataBuffer() {
+    for (uint64_t i = current_size; i > 0; --i) {
+        (buf + i - 1)->~T();
+    }
+    if (buf) {
+        std::free(buf);
+    }
+}
+
+template <class T>
+void DataBuffer<T>::resize(size_t new_size) {
+    if (new_size > current_capacity || !buf) {
+        if (buf) {
+            T* buf_old = buf;
+            buf = reinterpret_cast<T*>(std::malloc(sizeof(T) * new_size));
+            memcpy(buf, buf_old, sizeof(T) * current_size);
+            std::free(buf_old);
+        } else {
+            buf = reinterpret_cast<T*>(std::malloc(sizeof(T) * new_size));
+        }
+        current_capacity = new_size;
+    }
+    current_size = new_size;
+}
+
 template class DataBuffer<bool>;
```

Comment:
"explicit template instantiation" is definition and should be in source file (translation unit), or else in theory it's ODR violation. In practice it doesn't make sense to have them in header files (outline defined) which slows down compilation and leads to problems you're trying to work around.